### PR TITLE
Api get books of user donated

### PR DESCRIPTION
**Pull request by at-minhdao**
## Description
- Api get books of user donated
## Reference Requirements
- Product Backlog ID: None
- Task ID: None

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- None

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)
```mysql
select `books`.`id`, `books`.`title`, `books`.`description`, `books`.`language`,
     `books`.`rating`, `books`.`total_rating`, `books`.`picture`, `books`.`author`, 
    `books`.`price`, `books`.`unit`, `books`.`year`, `books`.`page_number` 
from `books` inner join `users` on `users`.`employ_code` = `books`.`from_person` 
where `users`.`id` = 3 and `books`.`deleted_at` is null 
order by `books`.`created_at` desc
```
```
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: users
   partitions: NULL
         type: const
possible_keys: PRIMARY,employ_code
          key: PRIMARY
      key_len: 4
          ref: const
         rows: 1
     filtered: 100.00
        Extra: Using filesort
*************************** 2. row ***************************
           id: 1
  select_type: SIMPLE
        table: books
   partitions: NULL
         type: ref
possible_keys: books_from_person_foreign
          key: books_from_person_foreign
      key_len: 42
          ref: const
         rows: 3
     filtered: 10.00
        Extra: Using index condition; Using where
```
- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: app/Http/Controllers/Api/UserController.php [hunk 1]
```diff
@@ -41,4 +42,37 @@ public function show($id)
 
         return $this->responseSuccess($user);
     }
+
+    /**
+     * Api get books of user donateds
```
Comment (at-trampham) [inline app/Http/Controllers/Api/UserController.php]: `donateds` dư `s` nha bạn :D

File: app/Http/Controllers/Api/UserController.php [hunk 2]
```diff
@@ -41,4 +42,37 @@ public function show($id)
...
+     *
+     * @param integer $id id of user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function getDonated(int $id)
```
Comment (at-trampham) [inline app/Http/Controllers/Api/UserController.php]: mình nghĩ là đặt tên `getBooksDonated` sẽ rõ ràng hơn

File: routes/api.php [hunk 3]
```diff
@@ -23,6 +23,7 @@
     Route::get('books/{book}', 'BookController@show');
     Route::get('books', 'BookController@index');
     Route::get('users/{user}', 'UserController@show');
+    Route::get('users/{user}/donated', 'UserController@getDonated');
```
Comment (at-trampham) [inline routes/api.php]: bỏ vào `middleware` cho security nha

**Review by at-trampham — COMMENTED** [review]
LGTM

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**